src/options.rs

```diff
@@ -208,10 +192,65 @@ impl Orientation {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ColorInversion {
+    Normal,
+    Inverted,
+}
+
+impl Default for ColorInversion {
+    fn default() -> Self {
+        Self::Normal
+    }
+}
+
+impl From<bool> for ColorInversion {
+    fn from(val: bool) -> Self {
+        match val {
+            true => Self::Inverted,
+            false => Self::Normal,
+        }
+    }
+}
+
+impl DcsCommand for ColorInversion {
+    fn instruction(&self) -> Instruction {
+        match self {
+            Self::Normal => Instruction::INVOFF,
+            Self::Inverted => Instruction::INVON,
+        }
+    }
+
+    fn fill_params_buf(&self, _buffer: &mut [u8]) -> Result<usize, Error> {
+        Ok(0)
+    }
+}
+
+///
+/// Display refresh order, defaults to left to right, top to bottom
+///
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum RefreshOrder {
+    /// Left to Right and Top to Bottom
+    Normal,
+    /// Right to Left and Top to Bottom
+    RightToLeft,
+    /// Left to Right and Bottom to Top
+    BottomToTop,
+    /// Right to Left and Bottom to Top
+    Inverted,
+}
```

Comment:
Hmm, I'm not sure I like the combination of these settings into a single enum. The first time I looked at the values I didn't understand what `Normal` and `Inverted` mean, which only became apparent when I looked at the comments.

This might also be a candidate for inclusion in `display-driver-hal`. Something like this:
```rust
pub enum VerticalRefreshOrder {
    TopToBottom,
    BottomToTop,
}

pub enum HorizontalRefreshOrder {
    LeftToRight,
    RightToLeft,
}

pub struct RefreshOrder {
    vertical: VerticalRefreshOrder,
    horizontal: HorizontalRefreshOrder,
}

impl RefreshOrder {
    pub const fn new(vertical: VerticalRefreshOrder, horizontal: HorizontalRefreshOrder) -> Self;
    pub const fn flip_horizontal(self) -> Self {...}
    pub const fn flip_vertical(self) -> Self {...}
}

// usage:
builder.with_refresh_order(RefreshOrder::default())...
builder.with_refresh_order(RefreshOrder::default().flip_vertical())...
builder.with_refresh_order(RefreshOrder::new(
    VerticalRefreshOrder::BottomToTop, HorizontalRefreshOrder::RightToLeft
))...
```

Reply:
done